src/message_queue.rs

```diff
@@ -65,4 +66,34 @@ impl MessageQueue {
         };
         Ok(queue)
     }
+
+    /// Retrieve (and remove) all messages of a given type from a given sender from the MessageQueue.
+    pub(crate) fn retrieve(
+        &mut self,
+        message_type: MessageType,
+        identifier: Identifier,
+        sender: ParticipantIdentifier,
```

Comment:
If this is an `Option`, is there a way to avoid each caller having to either add a `None` or wrap the `ParticipantIdentifier` in `Some()`?

Alternatively, is the later option as memory efficient? Or do you lose something by passing the full unfiltered Vec as a return?